apps/dashboard/src/components/side-navigation/constants.ts

```diff
@@ -8,79 +8,86 @@ import {
   RiStore3Line,
   RiUserAddLine,
 } from 'react-icons/ri';
+import type { IEnvironment } from '@novu/shared';
 import { NavItemsGroup } from './types';
-import { LEGACY_ROUTES, ROUTES } from '@/utils/routes';
+import { buildRoute, LEGACY_ROUTES, ROUTES } from '@/utils/routes';
 
-export const navigationItems: NavItemsGroup[] = [
-  {
-    items: [
-      {
-        label: 'Workflows',
-        icon: RiRouteFill,
-        to: ROUTES.WORKFLOWS,
-      },
-      {
-        label: 'Subscribers',
-        icon: RiGroup2Line,
-        isExternal: true,
-        to: 'https://docs.novu.co/api-reference/subscribers/get-subscribers',
-        disabled: true,
-      },
-    ],
-  },
-  {
-    label: 'Monitor',
-    items: [
-      {
-        label: 'Activity Feed',
-        icon: RiBarChartBoxLine,
-        to: LEGACY_ROUTES.ACTIVITY_FEED,
-        isExternal: true,
-      },
-    ],
-  },
-  {
-    label: 'Developer',
-    items: [
-      {
-        label: 'Integration Store',
-        icon: RiStore3Line,
-        to: LEGACY_ROUTES.INTEGRATIONS,
-        isExternal: true,
-      },
-      {
-        label: 'API Keys',
-        icon: RiKey2Line,
-        to: LEGACY_ROUTES.API_KEYS,
-        isExternal: true,
-      },
-    ],
-  },
-  {
-    label: 'Application',
-    items: [
-      {
-        label: 'Branding',
-        icon: RiPaintBrushLine,
-        to: LEGACY_ROUTES.BRANDING,
-        isExternal: true,
-      },
-      {
-        label: 'Settings',
-        icon: RiSettings4Line,
-        to: LEGACY_ROUTES.SETTINGS,
-        isExternal: true,
-      },
-    ],
-  },
-  {
-    items: [
-      {
-        label: 'Invite teammates',
-        icon: RiUserAddLine,
-        to: LEGACY_ROUTES.INVITE_TEAM_MEMBERS,
-        isExternal: true,
-      },
-    ],
-  },
-];
+export const buildNavigationItems = ({
+  currentEnvironment,
+}: {
+  currentEnvironment?: IEnvironment | null;
```

Comment:
Null shouldn't be allowed as a type. This method must be called only when the env is set. This will also simplify all the checks in the method.

Reply:
To ensure that, we can do exactly what we did in the current `web` app, introduce a Guard component that doesn't allow rendering of the layout until Clerk + Env are resolved.

Reply:
@SokratisVidros, IMO, it creates a bad UX, especially since it will wait for two requests to complete. It would be better to render some loading indicator/skeleton or smth; I mentioned this here:  https://novu.slack.com/archives/C07HVLG7G2U/p1728060743365279

Reply:
I agree 100% with the need for a splash/loading screen. That's exactly what the guard does in `web`. We "server-side" render the loader via the public.html and then it is removed only when env is resolved. If not mistaken we are saying the same thing.

Reply:
> If not mistaken we are saying the same thing
Not really, what I'm saying is to render the application shell while environments are loading: header, side nav, and in the main content, show the loading indicator. We can think about how it should look design-wise. I want to avoid empty screens, loading, blinking, etc., we can do better.

Reply:
I am aligned with that. The key is not to make the app interactive until Clerk + Env is loaded.

Reply:
The UX about the environment is different from allowing the `null` type. I still suggest to remove the `null` bit and prevent /env/undefined route generation.

Reply:
@SokratisVidros, we will solve it with UI state for the side nav while the env's are loading. Otherwise, I have to do the `if` statement and return nothing, so there will be a blinking experience